### ISSUE TITLE
It doesns't work as of June 2023

### ISSUE THREAD

**Issue by DATEx2** [AveYo/MediaCreationTool.bat#272]
It says Sorry we cannot determine at this time and then closes. This make your tool 100% useless
![image](https://github.com/AveYo/MediaCreationTool.bat/assets/83930915/fc68e5b1-2ac6-4bf2-a271-72b6170dafe6)

**Comment by rpodric** [comment]
https://github.com/AveYo/MediaCreationTool.bat/issues/252

**Comment by AveYo** [comment]
Sorry for being away for such a long time.  
Dev builds have been rebased. 25xxx for Canary, 23xxx Dev, 22000 Beta  
I did not anticipate a rebase, was thinking they are going to use 26xxx, so the code had this detection line: `if %VER% == 11 findstr /r "P.r.o.d.u.c.t.V.e.r.s.i.o.n...1.0.\..0.\..2.[256]"`  
The fix is simple, just had to add the 23xxx builds to that findstr regex:  
`if %VER% == 11 findstr /r "P.r.o.d.u.c.t.V.e.r.s.i.o.n...1.0.\..0.\..2.[235]"`
Will update the script as soon as I can.

**Comment by AveYo** [comment]
> I was able to upgrade to Windows 11 (21H2) from Windows 10 (22H2) under Apple BootCamp environment on July 3.
Probably. The issue only affected upgrading to 23xxx Dev builds.

**Comment by JustFragger** [comment]
@AveYo Hi, fix worked, now it says Windows Server installation, but when I continue, It throws again we have a problem determining whether windows server can be run on your computer.

Any tips?

**Comment by AveYo** [comment]
> Any tips?

run [windows_update_refresh.bat](https://github.com/AveYo/MediaCreationTool.bat/blob/main/bypass11/windows_update_refresh.bat)

**Comment by JustFragger** [comment]
> > Any tips?
> 
> run [windows_update_refresh.bat](https://github.com/AveYo/MediaCreationTool.bat/blob/main/bypass11/windows_update_refresh.bat)

Yeah I read readme in bypass11 and did that, but didn't worked. Same error.

My Windows 10 is bit broken, I can't fix my updated, after I download update and try to install it (restart and update) it throws we couldn't complete updates undoing changes. So maybe it's because of this I can't pass next.

**Comment by AveYo** [comment]
In that scenario an upgrade from 10 to 11 does not help.  
What you need is a successful repair-install with the current OS version and only after try upgrading.

Some more tips:  
- make sure there's at least 20GB free space on C  
- uninstall any 3rd party security suite (revert to Defender)  
- uninstall as many Store apps as you can  
for those that you can't there's [Edge_Removal.bat](https://github.com/AveYo/fox/blob/main/Edge_Removal.bat) and generic [FORCE APPX UNINSTALL.bat](https://pastebin.com/A48aF1BF)
- uninstall any Edge redirector (including my [OpenWebSearch.cmd](https://github.com/AveYo/fox/blob/main/OpenWebSearch.cmd))
- delete `C:\Windows.old` if present
- run [windows_update_cleanup.bat](https://pastebin.com/dKn5wiGj) and press `Y` to ResetBase when asked
- finally, run [windows_update_refresh.bat](https://github.com/AveYo/MediaCreationTool.bat/blob/main/bypass11/windows_update_refresh.bat)

I just upgraded a 10 to 11 22H2, then to 11 Dev via windows update (unsupported pc). 
First time it failed and undone it back to 11 22H2 - then I tried again with the steps above and 11 Dev went through _(sometimes it fails because it can't estimate required disk space correctly, other times because compatibility check can cause a race-condition, rarely is a store app / registry / permissions that breaks setup)_

**Comment by ElliotKillick** [comment]
If you're okay with WSL then I've just made a similar project called Mido: https://github.com/ElliotKillick/Mido

It's minimalist in that it's only for downloading the ISOs. Very clean and readable shell script. :)

Just an alternative for those who want something a bit simpler for VMs and such. Doesn't replace this project in any way.
